### DISCUSSION TITLE
MPlayer/mpv has finished unexpectedly. Exit code: 2

### DISCUSSION THREAD

**Discussion by I-am-Shibly** [smplayer-dev/smplayer#452] [Q&A] [answered]
I'm using the latest version of SMPlayer using **snapd**. Whenever I want to run **youtube** video, I find a popup saying the error that mentioned above. Here is the logs:

`/snap/smplayer/58/usr/bin/mpv --no-quiet --terminal --no-msg-color --input-ipc-server=/tmp/smplayer-mpv-48b4 --msg-level=ffmpeg/demuxer=error --video-rotate=no --no-fs --hwdec=no --sub-auto=fuzzy --vo=x11, --ao=pulse, --no-input-default-bindings --input-vo-keyboard=no --no-input-cursor --cursor-autohide=no --no-keepaspect --wid=106954848 --monitorpixelaspect=1 --osd-level=1 --osd-scale=1 --osd-bar-align-y=0.6 --sub-ass --embeddedfonts --sub-ass-line-spacing=0 --sub-scale=1 --sub-font=Arial --sub-color=#ffffffff --sub-shadow-color=#ff000000 --sub-border-color=#ff000000 --sub-border-size=0.75 --sub-shadow-offset=2.5 --sub-font-size=50 --sub-bold=no --sub-italic=no --sub-margin-y=8 --sub-margin-x=20 --sub-codepage=ISO-8859-1 --sub-pos=100 --vf-add=@veq:lavfi=[eq=1:0:1:1,hue=h=0] --volume=110 --cache=auto --screenshot-template=cap_%F_%p_%02n --screenshot-format=jpg --screenshot-directory=/home/samiul/Pictures/smplayer_screenshots --af-add=lavfi=[acompressor] --audio-pitch-correction=yes --volume-max=110 --ytdl -ytdl-format=bestvideo[height<=?360]+bestaudio/best[height<=?360] --script-opts=ytdl_hook-ytdl_path=yt-dlp --term-playing-msg=MPV_VERSION=${=mpv-version:}
INFO_VIDEO_WIDTH=${=width}
INFO_VIDEO_HEIGHT=${=height}
INFO_VIDEO_ASPECT=${=video-params/aspect}
INFO_VIDEO_FPS=${=container-fps:${=fps}}
INFO_VIDEO_FORMAT=${=video-format}
INFO_VIDEO_CODEC=${=video-codec}
INFO_DEMUX_ROTATION=${=track-list/0/demux-rotation}
INFO_AUDIO_FORMAT=${=audio-codec-name}
INFO_AUDIO_CODEC=${=audio-codec}
INFO_AUDIO_RATE=${=audio-params/samplerate}
INFO_AUDIO_NCH=${=audio-params/channel-count}
INFO_LENGTH=${=duration:${=length}}
INFO_DEMUXER=${=current-demuxer:${=demuxer}}
INFO_SEEKABLE=${=seekable}
INFO_TITLES=${=disc-titles}
INFO_CHAPTERS=${=chapters}
INFO_TRACKS_COUNT=${=track-list/count}
METADATA_TITLE=${metadata/by-key/title:}
METADATA_ARTIST=${metadata/by-key/artist:}
METADATA_ALBUM=${metadata/by-key/album:}
METADATA_GENRE=${metadata/by-key/genre:}
METADATA_DATE=${metadata/by-key/date:}
METADATA_TRACK=${metadata/by-key/track:}
METADATA_COPYRIGHT=${metadata/by-key/copyright:}
INFO_MEDIA_TITLE=${=media-title:}
INFO_STREAM_PATH=${stream-path}
 --audio-client-name=SMPlayer --term-status-msg=STATUS: ${=time-pos} / ${=duration:${=length:0}} P: ${=pause} B: ${=paused-for-cache} I: ${=core-idle} VB: ${=video-bitrate:0} AB: ${=audio-bitrate:0} https://www.youtube.com/watch?v=CCb_XbmB_iE

[ytdl_hook]  
[ytdl_hook] youtube-dl failed: not found or not enough permissions 
Failed to recognize file format.
Exiting... (Errors when loading file)
`
What should I do now?

**Comment by smplayer-dev** [comment, marked as answer]
Try the option "Install / Update Youtube support" from the Help menu.

**Reply by I-am-Shibly** [reply]
Thanks a lot. It solved my issue.
